### DISCUSSION TITLE
Which meshing software is used to generate Rotor 37 mesh in the tutorials?

### DISCUSSION THREAD

**Discussion by jh36714753** [mdolab/dafoam#347] [Tutorials]
Hello, I would like to know which meshing software is used to generate the Rotor 37 mesh in the tutorials.

I tried very hard to run flow simulation of Rotor 37 in DAFoam, using my own mesh which generated by turbogird. The same boundary condition and initial condition as tutorial. And the mesh quality is fine in turbogird and OpenFOAM meshCheck. But negative temperature occurred during calculation.

I really hope to get your help. Thank you.

**Comment by friedenhe** [comment]
We use ICEM to generate the mesh. Yes, DAFoam is currently not very robust for transonic turbomachinery cases. If you use totalPressure and totalTemperature as the inlet BC for this case, it should improve the simulation robustness. But DAFoam does not support totalPressure and totalTemperature BC, which means the adjoint's accuracy will be bad...

**Reply by jh36714753** [reply]
Thanks! That is very helpful.
